metadata-db-app/src/cmr/metadata_db/services/generic_documents.clj

```diff
@@ -21,53 +21,51 @@
   "Take the raw database object returned from either SQL or In-Memory and shape
    it into the expected response by pulling out generated fields like concept-id
    and putting them in the :Info field, then putting the metadata inside the
-   :Metadata field without any of these extra fields."
+   :Metadata field."
   [raw-doc]
-  ;; TODO: Generic work: if the lower levels use :extra-fields instead of mixing, then would this code be simplified?
-  {:Info (select-keys raw-doc info-fields)
-   :Metadata (reduce #(dissoc %1 %2) raw-doc fields-to-drop)})
+  {:Info (select-keys raw-doc info-fields) :Metadata (:metadata raw-doc)})
 
 (defn insert-generic-document
   "Insert a document under the provided provider-id. Generate a concept ID for
    the new record, At this time, nothing prevents multiple copies of a record
-   from being inserted, users must know what they are doing."
+   from being inserted, users must know what they are doing. Update will not call
+   this function."
   [context params provider-id raw-native-id document]
   (let [db (mdb-util/context->db context)
         ;; Validate that the provider exists.
         _ (provider-service/get-provider-by-id context provider-id true)
         document (if (map? document) (json/generate-string document) document)
         document-as-map (json/parse-string document true)
-        ;; TODO: Generic work: Fix native id so that it is passed in correctly.
         native-id (or raw-native-id (.toString (java.util.UUID/randomUUID))) ;; can this stay?
+        inner-metadata (json/parse-string (:metadata document-as-map) true)
+        doc-name (get inner-metadata :Name native-id)
         concept-type (get common-concepts/concept-prefix->concept-type (:concept-sub-type document-as-map))
         document-add (assoc document-as-map
                             :provider-id (str provider-id)
                             :concept-type concept-type
-                            :revision-id 1
+                            :document-name (subs doc-name 0 (min 20 (count doc-name)))
+                            :revision-id 1 ;; this function only for Create, not Update
                             :created-at (str (tkeeper/now))
                             :revision-date (dtp/clj-time->date-time-str (tkeeper/now))
                             :native-id native-id)
         concept-id (data/generate-concept-id db document-add)
         metadata (assoc document-add :concept-id concept-id)
-        _ (data/save-concept db provider-id metadata)
-        ;; TODO: Generic work: I think this is going to cause a race condition! We should return the actual thing that was saved and not get it back.
-        saved (first (data/get-latest-concepts db nil {:provider-id provider-id} [concept-id]))]
+        _ (data/save-concept db provider-id metadata)]
     (ingest-events/publish-event
      context
      (ingest-events/concept-update-event metadata))
-    (raw-generic->response saved)))
+    success-response))
```

Comment:
You either need to delete the raw-generic->response function because it isn't used anymore, or (and more preferred) to remove the success-response of nil and return a response from the DB.

Reply:
made a change to match concepts.